### PR TITLE
Add opencode as installer target with command generation and structure verification

### PR DESCRIPTION
**Pull request by heruujoko**
## Summary

Add OpenCode as a fourth install target alongside Codex, Claude, and Antigravity.

## Changes

- **New target `opencode`** in supported targets, usage string, flag description, and interactive prompt
- **`resolveOpenCodeDir()`** — searches for existing opencode config directory in priority order:
  1. `$OPENCODE_HOME` environment variable
  2. `$XDG_CONFIG_HOME/opencode`
  3. `~/.config/opencode`
  4. `~/.opencode`
  - Falls back to `~/.config/opencode` if none exist
- **`installOpenCodeCommands()`** — converts mothership skills to opencode commands:
  - `SKILL.md` in directories → `commands/<dir>.md` (e.g., `skills/commit/SKILL.md` → `commands/commit.md`)
  - Standalone `.md` in `skills/` root → `commands/<name>.md`
  - Nested skills preserved (e.g., `skills/obra/creating-plan/SKILL.md` → `commands/obra/creating-plan.md`)
  - `.system/` internal skills excluded
- **`verifyOpenCodeStructure()`** — post-install validation:
  - Confirms `commands/` directory exists
  - Verifies `.md` files are present
  - Checks no empty command files
- Standard dirs (`skills/`, `agents/`, `mothership-config/`) installed as-is alongside commands

## Test plan

- [x] All existing tests pass
- [x] New tests for `installOpenCodeCommands()` verify command generation from SKILL.md and standalone .md files
- [x] New tests for `verifyOpenCodeStructure()` cover valid, missing, and empty cases
- [x] New tests for `resolveOpenCodeDir()` validate env var priority and fallback behavior

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tools/installer/main.go [hunk 1]
```diff
@@ -232,11 +246,41 @@ func targetRootFor(target string) string {
...
+	for _, candidate := range searchPaths {
+		if info, err := os.Stat(candidate); err == nil && info.IsDir() {
+			return candidate
+		}
+	}
+
+	if xdg := os.Getenv("XDG_CONFIG_HOME"); xdg != "" {
+		return filepath.Join(xdg, "opencode")
+	}
+	return filepath.Join(home, ".config", "opencode")
```
Comment (heruujoko) [inline tools/installer/main.go:281]: ⚠️ **`filepath.Join` with `filepath.Dir` produces OS-specific separators**

On this line, `dir` comes from `filepath.Dir(rel)` which uses OS-specific path separators. On Windows, `dir` could be `obra\creating-plan`. Then `dir + ".md"` would produce `obra\creating-plan.md`, which would not create the correct nested structure via `filepath.Join(commandsDir, cmdRel)`.

**Suggestion:** Normalize `dir` to forward slashes before concatenation, or use `strings.ReplaceAll(dir, string(filepath.Separator), "/")`:

```go
// Normalize to forward slashes for consistent command paths
cmdRel = strings.ReplaceAll(dir, string(filepath.Separator), "/") + ".md"
```

This ensures that on all platforms, `filepath.Join(commandsDir, cmdRel)` produces the correct nested directory structure.

*— AI Review (Claude)*

**Review by heruujoko — COMMENTED** [review]
## 🔍 PR Review — Add OpenCode as Installer Target

**Reviewed by AI Agent (Claude)**

Overall, this is a well-structured PR that cleanly adds OpenCode as a fourth install target. The code follows the existing patterns, test coverage is solid, and the PR description clearly documents the changes. Below are a few issues I spotted.

### Summary of Findings

| Severity | Count |
|----------|-------|
| ⚠️ Important | 2 |
| 💡 Minor/Improvement | 1 |

### ✅ What's Good
- Clean separation of concerns with `resolveOpenCodeDir()`, `installOpenCodeCommands()`, and `verifyOpenCodeStructure()`
- Good XDG config directory precedence logic
- Proper `.system` exclusion for internal skills
- Test coverage for all three new functions including edge cases
- Consistent error wrapping with `fmt.Errorf` and `%w`

### ⚠️ Issues Found
See inline comments below for details on the three findings.

---
*This review was generated by an AI agent (Claude). Please use your judgment when addressing the feedback.*

File: tools/installer/main.go [hunk 2]
```diff
@@ -232,11 +246,41 @@ func targetRootFor(target string) string {
...
+	if xdg := os.Getenv("XDG_CONFIG_HOME"); xdg != "" {
+		searchPaths = append(searchPaths, filepath.Join(xdg, "opencode"))
+	}
+	searchPaths = append(searchPaths,
+		filepath.Join(home, ".config", "opencode"),
+		filepath.Join(home, ".opencode"),
+	)
+
+	for _, candidate := range searchPaths {
+		if info, err := os.Stat(candidate); err == nil && info.IsDir() {
```
Comment (heruujoko) [inline tools/installer/main.go:273]: ⚠️ **`isSupportedTarget` uses string equality chain — fragile for future targets**

Not critical, but `isSupportedTarget()` grows linearly with each new target. Consider using a map or slice-based lookup:

```go
var supportedTargets = map[string]bool{
    "codex": true, "claude": true, "antigravity": true, "opencode": true,
}

func isSupportedTarget(target string) bool {
    return supportedTargets[target]
}
```

This makes it O(1) and easier to maintain as targets grow. (Same consideration for `isSupportedMode`.)

*— AI Review (Claude)*

File: tools/installer/main.go [hunk 3]
```diff
@@ -344,9 +388,113 @@ func copyFileWithMode(src, dst string, mode os.FileMode) error {
...
+
+		count++
+		return nil
+	})
+	if err != nil {
+		return 0, fmt.Errorf("walk skills for opencode commands: %w", err)
+	}
+
+	return count, nil
+}
```
Comment (heruujoko) [inline tools/installer/main.go:458]: 💡 **Error return from `filepath.Walk` is discarded**

```go
filepath.Walk(commandsDir, func(path string, fi os.FileInfo, walkErr error) error {
```

The error return value of `filepath.Walk` is silently ignored. If the walk fails (e.g., permission error), `cmdFiles` could be incomplete and the verification would incorrectly report "no command files found" instead of the real error.

**Suggestion:**
```go
if err := filepath.Walk(commandsDir, func(...)); err != nil {
    return fmt.Errorf("walk commands dir: %w", err)
}
```

*— AI Review (Claude)*

File: tools/installer/main.go (hunk 1 above)
Comment (heruujoko) [reply tools/installer/main.go:281]: Fixed in fccf355. Replaced filepath.Join/filepath.Dir with path.Join/path.Dir for consistent forward-slash paths.

File: tools/installer/main.go (hunk 2 above)
Comment (heruujoko) [reply tools/installer/main.go:273]: Fixed in fccf355. Replaced `||` chains with `map[string]bool` lookups for both `isSupportedTarget` and `isSupportedMode`.

File: tools/installer/main.go (hunk 3 above)
Comment (heruujoko) [reply tools/installer/main.go:458]: Fixed in fccf355. Captured `filepath.Walk` error return and wrapped it with `fmt.Errorf("walk commands dir: %w", err)`.
